server/src/request.rs

```diff
@@ -65,52 +92,60 @@ pub fn process_analysis_request(request: AnalysisRequest) -> AnalysisResponse {
     // have a valid checksum, we return an error.
     for rule in &server_rules_to_rules {
         if !rule.verify_checksum() {
-            eprintln!("Rule {} has an invalid checksum", rule.name);
+            tracing::info!(
+                "Validation error: request rule `{}` has invalid checksum",
+                rule.name
+            );
             return AnalysisResponse {
                 rule_responses: vec![],
                 errors: vec![ERROR_CHECKSUM_MISMATCH.to_string()],
             };
         }
     }
 
+    let rules_count = rules.len();
+    let rules_str = if rules_count == 1 { "rule" } else { "rules" };
+    let rules_list = rules
+        .iter()
+        .map(|r| r.name.as_str())
+        .collect::<Vec<&str>>()
+        .join(", ");
     // execute the rule. If we fail to convert, return an error.
-    match rules_converted {
```

Comment:
Are we sure that we don't need this match here anymore?

Reply:
This is being addressed on line 74.